### ISSUE TITLE
Shapefile in non UTF-8 codepage problems

### ISSUE THREAD

**Issue by grassu** [MapWindow/MapWinGIS#124]
Hi,
I have just updated MW from version 4.9.4 to version 4.9.6. In  In opposite to the old version in 4.9.6 all shape-/dbf files seems to be treated as UTF-8. Now umlauts are no longer displayed correctly.

Is there a way to open a shapefile with an alternative codepage? An accompanying codepage file (*.cpg) seems not be considered.

Thank you very much,
grassu

**Comment by jerryfaust** [comment]
Hello @grassu 
This is very curious.  We made some changes to the Shapefile functionality in 4.9.6, adding support for new data types and linking the library directly into MapWinGIS (rather than calling into GDAL).  However, the base Shapefile functionality, as implemented in dbfopen.c and shpopen.c, are still the same files that are included in the GDAL distribution.  dfbopen.c is where CFG files are taken into consideration.

Are you able to send your shapefile, including the cfg file, I could try stepping through the load, and may be able to see what is going on.

Regards.
Jerry.

**Comment by grassu** [comment]
Dear Jerry,

thanks for your reply. See attached the shapefile in codepage 1252 (does not work in MW496) and utf8 (works in MW496).
I tested it on a german windows 7 and windows 2010.

Regards,
grassu

[test-1252.zip](https://github.com/MapWindow/MapWinGIS/files/2472164/test-1252.zip)
[test-utf8.zip](https://github.com/MapWindow/MapWinGIS/files/2472165/test-utf8.zip)

**Comment by jerryfaust** [comment]
Hello @grassu 

I was making changes that were related to your issue, and went back to retest your data.  Is it possible that the data you submitted is actually identical? in other words, having identical .dbf files, but having different code page files?  After trying various things, I was not getting the behavior I was expecting, specifically from the UTF-8 database.  So I tried just changing the .cpg file for the UTF-8 database - changing it to 1252- and the data then loaded and displayed properly.  This leads me to believe that the data is not in UTF-8 format, but instead is actually in the Windows Latin 1 (code page 1252).

The short of it is that I believe (and hope) that the problem you were experiencing is now fixed (although for you to use it, you will have to get/build the latest 'develop' branch).

Let me know what you think.
Regards.
Jerry.
